annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -536,33 +587,117 @@ private TranscriptConsequenceSummary getCanonicalTranscript(VariantAnnotation gn
     }
 
     @Override
-    public List<AnnotatedRecord> getAnnotatedRecordsUsingPOST(AnnotationSummaryStatistics summaryStatistics, List<MutationRecord> mutationRecords, String isoformOverridesSource, Boolean replace) throws Exception {
+    public List<AnnotatedRecord> getAnnotatedRecordsUsingPOST(AnnotationSummaryStatistics summaryStatistics, List<MutationRecord> mutationRecords, String isoformOverridesSource, Boolean replace) {
+        // this will send everything at once
+        return getAnnotatedRecordsUsingPOST(summaryStatistics, mutationRecords, isoformOverridesSource, replace, mutationRecords.size());
+    }
+
+    @Override
+    public List<AnnotatedRecord> getAnnotatedRecordsUsingPOST(AnnotationSummaryStatistics summaryStatistics, List<MutationRecord> mutationRecords, String isoformOverridesSource, Boolean replace, Integer postIntervalSize) {
         // construct list of genomic location objects to pass to api client
+        // TODO use SortedSet (or at least Set) instead?  Do we anticipate a lot of redundancy? Probably quite a bit.
+        // Maybe test which is faster with a large study
```

Comment:
I think what's important here is to make sure we are retaining the order of the sample/variants submitted for annotation. As a user I would want to make sure that I would be able to expect sample 1-variant 1 to appear in the same row in the input MAF as it does in the annotated output MAF. 

As long as the implementation of a SortedSet or Set does not affect the ordering of the annotated records in the output file then I think this would be okay to do.

Reply:
My memory is that a map of <query> -> <annotation> is constructed with all successful annotations ... and then when the response is constructed we do so by iterating through the original query in the query order in order to select response records in the corresponding order (unless no answer was found ... which generates the "successfully_annotated:fase" records)